flight/Modules/Battery/battery.c

```diff
@@ -149,14 +155,21 @@ static void batteryTask(void * parameters)
 					scaled_voltage = 0.0f;
 					adc_offset_invalid = true;
 				}
+				else if(scaled_voltage > 3.0f && !cells_calculated && (batterySettings.MaxCellVoltage > 0)){
+					cells_calculated = true;
+					cells = ((scaled_voltage / batterySettings.MaxCellVoltage) + 0.9f);
+					flightBatteryData.DetectedCellCount = cells;
+				}
+				else if(!cells_calculated)
+					cells = 1;
 			}
 
 			flightBatteryData.Voltage = scaled_voltage;
 
 			// generate alarms and warnings
-			if (flightBatteryData.Voltage < batterySettings.VoltageThresholds[FLIGHTBATTERYSETTINGS_VOLTAGETHRESHOLDS_ALARM])
+			if (flightBatteryData.Voltage < (batterySettings.VoltageThresholds[FLIGHTBATTERYSETTINGS_VOLTAGETHRESHOLDS_ALARM] * cells))
```

Comment:
I think we need some way to turn this on and off.   Use cells specified in UI, or if 0, do autodetect.  Also change UI to indicate (per cell) for the voltages.

It probably also makes sense to rename these parameters so people upgrading don't get bogus settings.